src/Providers/ProvidersPage.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'src/internal/i18n';
+import { ProviderResource } from 'src/internal/k8s';
+
+import { MOCK_CLUSTER_PROVIDERS } from '@app/queries/mocks/providers.mock';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  Level,
+  LevelItem,
+  PageSection,
+  Title,
+  Toolbar,
+  ToolbarContent,
+  ToolbarToggleGroup,
+} from '@patternfly/react-core';
+import { FilterIcon } from '@patternfly/react-icons';
+
+import AttributeValueFilter from './components/AttributeValueFilter';
+import EnumFilter from './components/EnumFilter';
+import FreetextFilter from './components/FreetextFilter';
+import { ManageColumnsToolbar } from './components/ManageColumnsToolbar';
+import PrimaryFilters from './components/PrimaryFilters';
+import ProviderRow from './components/ProviderRow';
+import { createMetaMatcher, Field } from './components/shared';
+import { NAME, READY, TYPE, URL } from './components/shared';
+import TableView from './components/TableView';
+
+const isMock = process.env.DATA_SOURCE === 'mock';
+
+const useProviders = ({ kind, namespace }) => {
```

Comment:
we should consider some `moakcableUseK8sWatchResource` ( or some other name ) method that will incapsulate this logic to one line:
```ts
const [providers, loaded, error] = moakcableUseK8sWatchResource<ProviderResource[]>({
        kind,
        isList: true,
        namespaced: true,
        namespace,
      },
      [MOCK_CLUSTER_PROVIDERS, true, false] );
```

Reply:
Created `useMockableK8sWatchResource` helper function.